WireCells/Sources/WireCellsImplementation/NodesAPI/ObservableStream.swift

```diff
@@ -0,0 +1,97 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import Smithy
+
+/// Wraps a `Smithy.Stream` allowing its progress to be observed. This is a workaround to enable reading upload progress
+/// using the AWS SDK.
+
+final class ObservableStream: Smithy.Stream, Sendable {
+
+    private let wrapped: any Smithy.Stream
```

Comment:
minor: for a bit more readability
```suggestion
    private let wrappedStream: any Smithy.Stream
```

Reply:
I've updated this in [954b36d](https://github.com/wireapp/wire-ios/pull/3096/commits/954b36d3d0abb80e021cb97272f80d303833bfa8)